ntoskrnl/mm/ARM3/sysldr.c

```diff
@@ -2964,7 +2964,52 @@ MmLoadSystemImage(IN PUNICODE_STRING FileName,
     PrefixName = *FileName;
 
     /* Check if we have a prefix */
-    if (NamePrefix) DPRINT1("Prefixed images are not yet supported!\n");
+    if (NamePrefix)
+    {
+        /* Check if "directory + prefix" is too long for the string */
+        Status = RtlUShortAdd(PrefixName.MaximumLength,
+                              BaseDirectory.Length,
+                              &PrefixName.MaximumLength);
+        if (!NT_SUCCESS(Status))
+        {
+            Status = STATUS_INVALID_PARAMETER;
+            goto Quickie;
```

Comment:
In this case you are accessing an uninitialized pointer when you are calling `ExFreePoolWithTag(PrefixedBuffer, TAG_LDR_WSTR);`.
You should probably initialize `PrefixedBuffer` to `NULL`, and check that it is not null instead of checking for `NamePrefix` in the cleanup.

Reply:
Changes have been pushed.